lib/deploy-aws-assets.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  CfnParameter, Construct, Stack,
+} from '@aws-cdk/core';
+import { CiEcrStack, EcrStackProps } from './ci-ecr-stack';
+
+export interface DeployAssetsProps extends EcrStackProps {
+  /** Environment in which assets are being deployed */
+  readonly envName: string;
+}
+
+export class DeployAwsAssets extends Stack {
+  constructor(scope: Construct, id: string, props: DeployAssetsProps) {
+    super(scope, id, props);
+    const deployECRParameter = `${props?.deployECR ?? this.node.tryGetContext('deployEcr')}`;
+
+    new CfnParameter(this, 'deployEcr', {
+      description: 'Should ECR repositories + roles be created',
+      default: deployECRParameter === 'true',
```

Comment:
I think default should be false. These are optional AWS assets that is not essential to jenkins deployment

Reply:
The default is `false`. This will evaluate to `false` if the value is anything other than `true`

Reply:
Why not take it as cfn inputs that are passed as parameters like 'failOnCloudInitError' instead of getting it from context? Default it to false and if the customer needs they can add the value in props like this: https://github.com/opensearch-project/opensearch-ci/blob/main/lib/ci-stack.ts#L33-L35

Reply:
It rather mimics the behavior of `useSsl` and `runWithOidc`. https://github.com/opensearch-project/opensearch-ci/blob/main/lib/ci-stack.ts#L29-L31

Doing it like this gives the user to control the behavior from the command line rather than modifying the code to make the changes. This can always be passed as a parameter too if someone wants to do that as here - https://github.com/opensearch-project/opensearch-ci/pull/77/files/f3027a2459b8bd3ac3c23b30d1e341dba7fadc99#diff-d638b1fcda588e6ca1f48230caf1d17595dccafad1e580bcd937a15150343471R11

This line however is only to record the changes to the parameter to cloud formation and not take input as https://github.com/opensearch-project/opensearch-ci/blob/main/lib/ci-stack.ts#L67-L70

Reply:
Was wondering if we really need it to provide it as cli param. Keeping it default as false would work as well. Anyway its a nit, we can refactor later if this doesn't work out.